### ISSUE TITLE
Stratum data hashrate overestimated until 24 hours

### ISSUE THREAD

**Issue by Cyrix126** [SChernykh/p2pool#329]
The HR 15m/1h/24h values from the local/stratum and from the status console reports same the values for the different time frames.

I expect after the first minutes passed to have 15m > 1h > 24h.
Currently it is 15m == 1h == 24h.

After 15m, the 15m value will be different from 1h and 24h which will both be the same value.
But the values are wrong since it's been only a few minutes, it is not possible to have the same value for 1h and 24h.

So because of this, values will be over estimated for 15 minutes for 15m, 1 hour for 1h and 24 hours for 24h.

I can reproduce this bug on v4.0.

**Comment by SChernykh** [comment]
It is an intended behavior. If less than 24 hours passed, hashrate for "24h" will be counted a smaller time period. This is why all 3 hashrates are the same in the first 15 minutes, and so on.

**Comment by Cyrix126** [comment]
This is a bit confusing. I would expect an average for the timeframe. For example if it had 1kH/s for only 12h, to have about 500H/s for the 24h value, like xmrig is doing.

**Comment by SChernykh** [comment]
XMRig doesn't have 24h hashrate - it has 10s/1m/15m values and it just doesn't show 15m hashrate until 15 minutes have passed.

**Comment by Cyrix126** [comment]
> XMRig doesn't have 24h hashrate - it has 10s/1m/15m values and it just doesn't show 15m hashrate until 15 minutes have passed.

You are right. I was thinking of xmrig-proxy. It does behave the way I was describing AFAIK.

**Comment by SChernykh** [comment]
> > XMRig doesn't have 24h hashrate - it has 10s/1m/15m values and it just doesn't show 15m hashrate until 15 minutes have passed.
> 
> You are right. I was thinking of xmrig-proxy. It does behave the way I was describing AFAIK.
> 
> I guess I can make the calculation by tracking time and the value total_hashes from local/stratum. It would be fine for me.

Yes. And uptime can be found in local/p2p json.
